task/hiffy/src/common.rs

```diff
@@ -729,6 +731,56 @@ pub(crate) fn sprot_block_size(
     Ok(4)
 }
 
+#[cfg(feature = "sprot")]
+pub(crate) fn sprot_reset_component(
+    stack: &[Option<u32>],
+    data: &[u8],
+    _rval: &mut [u8],
+) -> Result<usize, Failure> {
+    if stack.len() < 3 {
+        return Err(Failure::Fault(Fault::MissingParameters));
+    }
+    let fp = stack.len() - 3;
+    let intent: drv_update_api::ResetIntent = match stack[fp + 0] {
+        // Some(intent) => match drv_update_api::ResetIntent::try_from(intent as u8) {
+        Some(intent) => {
+            match drv_update_api::ResetIntent::from_u8(intent as u8) {
+                Some(intent) => intent,
+                None => return Err(Failure::Fault(Fault::BadParameter(0))),
+            }
+        }
+        None => return Err(Failure::Fault(Fault::EmptyParameter(0))),
```

Comment:
FYI, all of these match/None/return sequences could be replaced by `.ok_or(error)?`.